hbase_sql/src/main/java/com/splicemachine/backup/BackupUtils.java

```diff
@@ -0,0 +1,211 @@
+package com.splicemachine.backup;
+
+import com.splicemachine.access.HConfiguration;
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.sql.dictionary.BackupDescriptor;
+import com.splicemachine.hbase.ZkUtils;
+import com.splicemachine.pipeline.Exceptions;
+import com.splicemachine.si.constants.SIConstants;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.regionserver.HRegion;
+import org.apache.hadoop.hbase.regionserver.HRegionFileSystem;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper;
+import org.apache.log4j.Logger;
+import org.apache.zookeeper.KeeperException;
+
+import java.io.IOException;
+
+/**
+ * Created by jyuan on 8/10/16.
+ */
+public class BackupUtils {
+    private static final Logger LOG=Logger.getLogger(BackupUtils.class);
+    /**
+     *
+     * @param fs HBase file system
+     * @param rootPath HBase root directory
+     * @return True if there exists a successful database backup
+     * @throws IOException
+     */
+    public static boolean existsDatabaseBackup(FileSystem fs, Path rootPath) throws IOException {
+        boolean ret = false;
+        FSDataInputStream in = null;
+        try {
+            // Open backup record file from file system
+            Path backupPath = new Path(rootPath, BackupRestoreConstants.BACKUP_DIR);
+            Path p = new Path(backupPath, BackupRestoreConstants.BACKUP_RECORD_FILE_NAME);
+            if (fs.exists(p)) {
+                in = fs.open(p);
+                int n = in.readInt();  // number of records
+                BackupDescriptor bd = new BackupDescriptor();
+                while (n-- > 0) {
+                    bd.readExternal(in);
+                    if (bd.getScope().compareToIgnoreCase("DATABASE") == 0) {
+                        ret = true;
+                        break;
+                    }
+                }
+            }
+            return ret;
+        }
+        finally {
+            if (in != null)
+                in.close();
+        }
+    }
+
+    /**
+     * Given a file path for an archived HFile, return the corresponding file path in backup metadata directory
+     * @param p file path for archived HFile
+     * @return orresponding file path in backup metadata directory
+     */
+    public static String getBackupFilePath(String p) {
+        return p.replace(BackupRestoreConstants.ARCHIVE_DIR, BackupRestoreConstants.BACKUP_DIR);
+    }
+
+    public static void waitForBackupToComplete(String tableName, String regionName, String path) throws IOException{
+        int i = 0;
+        long maxWaitTime = 60*1000;
+        while (regionIsBeingBackup(tableName, regionName, path)) {
```

Comment:
Do we need a final static constant there for maxWaitTime and take it off the method ?